packages/dataviews/src/components/dataviews/stories/index.story.tsx

```diff
@@ -184,3 +184,39 @@ export const CombinedFields = () => {
 		/>
 	);
 };
+
+export const PinnedItems = () => {
+	const [ view, setView ] = useState< View >( {
+		...DEFAULT_VIEW,
+		fields: [ 'title', 'description', 'categories' ],
+	} );
+	const { data: shownData, paginationInfo } = useMemo( () => {
+		return filterSortAndPaginate( data, view, fields );
+	}, [ view ] );
+	// Pin the first item
+	const [ pinnedItems, setPinnedItems ] = useState< string[] >( [
+		...shownData[ 0 ].id.toString(),
+	] );
+
+	return (
+		<DataViews
+			getItemId={ ( item ) => item.id.toString() }
+			paginationInfo={ paginationInfo }
+			data={ shownData }
+			enablePinnedItems
+			pinnedItems={ pinnedItems }
+			onPinItem={ ( itemId: string ) => {
```

Comment:
My suggestion is that [we don't allow users to pin/unpin](https://github.com/WordPress/gutenberg/pull/65629#issuecomment-2376574755) in this PR, so we can remove the `onPinItem`/`onUnpinItem` props and all related code.

However, it's interesting to think how that could work in the future to make sure we have the right APIs. I'm thinking that there's nothing special about pinning/unpinning: it's just another action over the dataset. It seems to me that we could model this behavior using the existing actions API — so that's another argument to not introducing these props.

Reply:
> However, it's useful to think how that could work in the future to make sure we have the right APIs. I'm thinking that there's nothing special about pinning/unpinning: it's just another action over the dataset. It seems to me that we could model this behavior using the existing actions API — so that's another argument to not introducing these props.

More thoughts: if, in the future, we want specific affordances for pinning like we have for selection, we may actually want to introduce specific callbacks. I think this would depend on whether or not we expect the layouts to implement a specific affordances for pinning (like we have for selection) vs just listing it as another action.

In any case, that's a future concern :)